### PR TITLE
load fewer models for the tests

### PR DESCRIPTION
**Pull request by MauriceHendrix**
## Description

Use conftest.py to load models and also reduce number of models loades

## Motivation and Context

wanted to speed up testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
Speed up testing somewhat

## Documentation checklist

- [ ] I have updated all documentation in the code where necessary.
- [ ] I have checked spelling in all (new) comments and documentation.
- [ ] I have added a note to RELEASE.md if relevant (new feature, breaking change, or notable bug fix).
No functional changes

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

File: chaste_codegen/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from chaste_codegen import DATA_DIR, load_model_with_conversions
+
+
+cellml_folder = os.path.join(DATA_DIR, 'tests', 'cellml')
+
+
+@pytest.fixture(scope='session')
+def s_model():
+    model_folder = os.path.join(cellml_folder, 'Shannon2004.cellml')
```
Comment (MichaelClerx) [inline chaste_codegen/tests/conftest.py:14]: That's not a folder :D

File: chaste_codegen/tests/test_linearity_check.py [hunk 2]
```diff
@@ -57,11 +47,17 @@ def test_wrong_params_get_non_linear_state_vars2(derivatives_eqs, membrane_volta
 
 def test_get_non_linear_state_vars(non_linear_state_vars):
     non_linear_state_vars = sorted(non_linear_state_vars, key=lambda s: Printer().doprint(s))
+    print(str(non_linear_state_vars))
```
Comment (MichaelClerx) [inline chaste_codegen/tests/test_linearity_check.py:50]: Debug code?

File: chaste_codegen/tests/test_linearity_check.py [hunk 3]
```diff
@@ -89,77 +85,62 @@ def test_wrong_params_subst_deriv_eqs7(y_derivatives, membrane_voltage_var, non_
...
-                        "tion$Ca_o) + 1.0)) + _calcium_background_current$g_b_Ca*(-0.5*_membrane$RTONF*log(_extracellu"
-                        "lar_calcium_concentration$Ca_o/_intracellular_calcium_concentration$Ca_i) + _membrane$V) + 4."
-                        "0*_second_inward_calcium_current$P_si*_second_inward_calcium_current_d_gate$d*(1.0 - _second_"
-                        "inward_calcium_current_f_Ca_gate$f_Ca)*(_membrane$V - 50.0)*(-_extracellular_calcium_concentr"
-                        "ation$Ca_o*exp_(-2.0*(_membrane$V - 50.0)/_membrane$RTONF) + _intracellular_calcium_concentra"
-                        "tion$Ca_i*exp_(100.0/_membrane$RTONF))*(-_intracellular_calcium_concentration$Ca_i/(_intracel"
-                        "lular_calcium_concentration$Ca_i + 0.001) + 1.0)/(_membrane$RTONF*(1.0 - exp_(-2.0*(_membrane"
-                        "$V - 50.0)/_membrane$RTONF))))/(_intracellular_calcium_concentration$Ve*_membrane$F))]")
+                                          s_model.get_equations_for)
+    assert str(deq) == \
```
Comment (MichaelClerx) [inline chaste_codegen/tests/test_linearity_check.py:92]: No easy way to make this more useful for debugging? Move the string into a file? Compare line by line?

File: chaste_codegen/tests/test_lookup_tables.py [hunk 4]
```diff
@@ -1,113 +1,184 @@
...
+             '3.5 * exp(-pow((cell$V / 30.0), 2.0)) + 1.5',
+             '1.0 / (1.0 + exp((33.5 + cell$V) / 10.0))',
+             '20.0 / (1.0 + exp((33.5 + cell$V) / 10.0)) + 20.0',
+             '1.0 / (1.0 + exp((-14.5 - cell$V) / 6.0))',
+             '-exp((-14.5 - cell$V) / 6.0) + 1.0',
+             '1.0 / (1.0 + exp((35.06 + cell$V) / 3.6)) + 0.6 / (1.0 + exp((-cell$V + 50.0) / 20.0))',
+             '1.0 / (0.0197 * exp(-pow(((14.5 + cell$V) * 0.0337), 2.0)) + 0.02)']}]
+
+
+def test_change_lookup_table(be_model):
```
Comment (MichaelClerx) [inline chaste_codegen/tests/test_lookup_tables.py:173]: Could all do with a lot more comments? If these tests stop working in 5 years will you remember what they were supposed to do?

**Review by MichaelClerx — APPROVED** [review]

File: chaste_codegen/tests/test_lookup_tables.py (hunk 4 above)
Comment (MauriceHendrix) [reply chaste_codegen/tests/test_lookup_tables.py:173]: Most of these I think so as I've tried to at least give them descriptive names.

File: chaste_codegen/tests/test_linearity_check.py (hunk 3 above)
Comment (MauriceHendrix) [reply chaste_codegen/tests/test_linearity_check.py]: Moving the strings to a file is a fair point

**Pull request merged**